test/Microsoft.AspNetCore.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs

```diff
@@ -1029,6 +1151,23 @@ public void ModelStateDictionary_AddsErrorMessage_ForInputFormatterException()
             var expectedMessage = "This is an InputFormatterException";
             var dictionary = new ModelStateDictionary();
 
+            // Act
+            dictionary.TryAddModelError("key", new InputFormatterException(expectedMessage));
+
+            // Assert
+            var entry = Assert.Single(dictionary);
+            Assert.Equal("key", entry.Key);
+            var error = Assert.Single(entry.Value.Errors);
+            Assert.Equal(expectedMessage, error.ErrorMessage);
```

Comment:
As mentioned above, I'm not a fan of this inconsistent case. But, regardless of the decision on the semantics, need to test with `InputFormatterException`. Assert about `error.Exception` here too (either `null` or identical to the passed-in instance depending on decisions)